WordPress/Classes/Networking/MediaServiceRemoteXMLRPC.m

```diff
@@ -40,6 +40,32 @@ - (void)getMediaWithID:(NSNumber *)mediaID
                  }];
 }
 
+/** Adds a basic auth header to a request if a credential is stored for that specific host.
+ 
+ The credentials will only be added if a set of credentials for the request host are stored on the shared credential storage
+ @param request, the request to where the authentication information will be added.
+ */
+- (void)addBasicAuthCredentialsIfAvailableToRequest:(NSMutableURLRequest *)request
+{
+    NSURLCredentialStorage *credentialStorage = [NSURLCredentialStorage sharedCredentialStorage];
+    NSInteger port = [[request.URL port] integerValue];
+    if (port == 0) {
+        port = 80;
+    }
+    NSURLProtectionSpace *protectionSpace = [[NSURLProtectionSpace alloc] initWithHost:[request.URL host]
+                                                                                  port:port
+                                                                              protocol:[request.URL scheme]
+                                                                                 realm:@"Restricted Area"
```

Comment:
@SergioEstevao It's possible that the `realm` will be named something different on another server so this code will probably break. As a test I changed the realm name on our sample site and I was not able to upload images anymore.